Adds Option to Retry Any Error Reason

ncronquist opened the pull request:
Hey Matt, 

First of all, thanks for the great library - simple and useful :)

Not sure if this PR is something your interested in or not, so feel free to close without comment, but we needed this change for some http errors we are seeing in our system like, `ehostunreach`, `econnrefused`, and `closed`, and I thought I'd offer the upstream change.

If you have any suggestions for doing this differently, I'd be open to make the changes too.

Thanks again! 

- Adds new option `retry_all_errors` to retry HTTPoison requests
  that fail with any error reason 
- Backwards compatible to only retry nxdomain and timeout
  errors by default
- Works just like the include_404s option

ncronquist committed:
Adds Option to Retry Any Error Reason

- Adds new option `retry_all_errors` to retry HTTPoison requests
  that fail with any error reason
- Backwards compatible to only retry nxdomain and timeout
  errors by default
- Works just like the include_404s option

ncronquist committed:
Add retry_all_errors to README

mgwidmann commented on lib/httpoison/retry.ex:
```diff
@@ -37,6 +37,7 @@ defmodule HTTPoison.Retry do
         max_attempts: Application.get_env(:httpoison_retry, :max_attempts) || unquote(@max_attempts),
         wait: Application.get_env(:httpoison_retry, :wait) || unquote(@reattempt_wait),
         include_404s: Application.get_env(:httpoison_retry, :include_404s) || false,
+        retry_all_errors: Application.get_env(:httpoison_retry, :retry_all_errors) || false,
```
Maybe rename to `retry_unknown_errors` since thats what is actually happening here. Then default to `true` (you'll have to change the `|| false` to `Application.get_env(:httpoison_retry, :retry_unknown_errors, true)`).

ncronquist replied:
Hey Matt, I renamed the option to `retry_unknown_errors`. 

You mentioned defaulting to `true` for that option, but I was concerned that wouldn't be backwards compatible with the original retry settings and I just wanted to make sure that's what you meant?

mgwidmann commented:
Sorry for the delay, I'd be happy to merge once we rename the parameter. Also, if theres any documentation on it we should probably update it.

Commit:
Renames Option to retry_unknown_errors

- Renames the option to retry errors not specifically defined as
  `retry_unknown_errors`

ncronquist commented:
I renamed the option as you requested and added a comment to your comment about defaulting that value to true.

As far as the documentation, I made sure it was noted in the same way as the `include_404s` option. Let me know if you would like something a bit more in depth.
